### ISSUE TITLE
Exception Peek UI should be placed on the top frame with source

### ISSUE THREAD

**Issue by gregg-miskelly** [Microsoft/vscode#25104]
- VSCode Version:
- OS Version:

Steps to Reproduce:
1. Create a new .NET Core project (ex: `dotnet new console` + `dotnet restore`)
2. Open the project in VS Code + C# extension
3. Add a line that will throw an exception from non-user code. For a trivial example add `String.Format(null, 12)` as the first line of main.
4. Start debugging

Result: 
Debugger will break on the exception, but it will not be clear where or why as VS Code will not bring up the top-most document that has source.

Expected:
VS Code should walk the frames of the stack and choose the first that has a `source` property. VS Code used to do this.

Example json response to the `stackTrace` command:

```json
{
    "seq":29,
    "type":"response",
    "request_seq":11,
    "success":true,
    "command":"stackTrace",
    "message":"",
    "body":{
        "stackFrames": [
            {"id":1000,"name":"[External Code]","line":0,"column":0},
            {
                "id":1001,
                "name":"clicon.dll!Program.Main(string[] args) Line 14",
                "source": {
                    "name":"Program.cs",
                    "path":"d:\\proj\\clicon\\Program.cs",
                    "sourceReference":0,
                    "checksums":[{"algorithm":"SHA1","checksum":"7241f6d2a81d39115752953170fa4e9391d5fcf0"}]
                },
                "line":14,
                "column":9,
                "endLine":14,
                "endColumn":33,
                "moduleId":"{7C382E61-E5C4-49EC-A609-69A65BF4E458}"
            },
            {"id":1002,"name":"[External Code]","line":0,"column":0}
        ],
        "totalFrames":3
    }
}
```

**Comment by gregg-miskelly** [comment]
@isidorn I am still able to reproduce this problem in current insiders. Should I open a new issue?

**Comment by isidorn** [comment]
@gregg-miskelly no need for a new issue if the steps are the same, let's reopen this one and I will investigate

**Comment by isidorn** [comment]
This issue was introduced due to some improvements in call [stack fetching](https://github.com/Microsoft/vscode/issues/25605)

With the latest fix the issue should be solved. However take note that the handshake mechanism is a pain for me to try out the latest vscode chagnes on the fly since my development version of vscode does not have the right handshake. And due to that I could not verify the fix, but I will retry it in tomorrow's insiders.

Thansk

**Comment by gregg-miskelly** [comment]
@isidorn I verified with today's insiders build (commit 64d532ccb681ddbef0039db1effa2949f1993214). The experience is much better now -- it will automatically navigate to the leaf frame that has source. But it still doesn't put up the exception peak window. I verified that the debug adapter don't get an `exceptionInfo` request in this case.

If it is helpful, I can privately share a version of the adapter that has the handshake disabled. Email me and let me know what OS you are on.
